lib/IDE/SyntaxModel.cpp

```diff
@@ -547,15 +522,31 @@ static bool shouldTreatAsSingleToken(const SyntaxStructureNode &Node,
              SM.getLineAndColumnInBuffer(Node.Range.getEnd()).first;
 }
 
+std::pair<bool, ArgumentList *>
+ModelASTWalker::walkToArgumentListPre(ArgumentList *ArgList) {
+  for (auto Arg : *ArgList) {
+    auto res = ArgumentInfo.insert({Arg.getExpr(), Arg});
```

Comment:
I guess this is going to store 4 extra pointers for every argument expression in a source file, which might be kind of big.  WDYT of keeping a counter of active argument lists, decremented in `walkToArgumentList`, and if it hits 0 we call `ArgumentInfo.clear()`?  That avoids the cost of removing individual expressions but keeps the map from growing more than necessary.

Reply:
Yeah, makes sense, how does this look?

Reply:
LGTM, thanks!